src/Auth0.ManagementApi.Client/ApiConnection.cs

```diff
@@ -125,6 +128,13 @@ public ApiConnection(string token, string baseUrl)
                 }
             }
 
+            // Add the diagnostics header, unless user explicitly opted out of it
+            if (!object.ReferenceEquals(diagnostics, DiagnosticsHeader.Suppress))
+                request.AddHeader("Auth0-Client", diagnostics.ToString());
+
+            // Add the user agent
+            request.AddHeader("User-Agent", ".NET/PCL");

```

Comment:
Yeah, I was thinking about the bait-and-switch technique too. We can look into it, but unfortunately I do not have access to the 2 Xamarin platforms, should be decide to go that route.

As far as the opt-in is concerned, I used the technique from the V1 library. Essenstially all you need to do is pass the `DiagnosticsHeader .Suppress` value to the constructor of `ManagementApiClient` and no diagnostics information will be sent at all.

Alternatively you can build up your own `DiagnosticsHeader` with whichever values you want, and I will send that along instead.